src/model/entity/enemy.go

```diff
@@ -24,10 +24,143 @@ const (
 	DirectionStop
 )
 
+type AttributeRate uint
+
+const (
+	AttributeRateLow      AttributeRate = 25
+	AttributeRateAverage  AttributeRate = 50
+	AttributeRateHigh     AttributeRate = 75
+	AttributeRateVeryHigh AttributeRate = 100
+)
+
+type HostilityRadius uint
+
+const (
+	HostilityRadiusLow     HostilityRadius = 2
+	HostilityRadiusAverage HostilityRadius = 4
+	HostilityRadiusHigh    HostilityRadius = 6
+)
+
 type Enemy struct {
-	BaseCharacter Character
-	Type          EnemyType
-	Hostility     uint
-	IsChasing     bool
-	Direction     Direction
+	Character       Character
+	Type            EnemyType
+	HostilityRadius HostilityRadius
+	IsChasing       bool
+	Direction       Direction
+}
+
+type Zombie struct {
+	Enemy Enemy
+}
+
+type Vampire struct {
+	Enemy          Enemy
+	HadFirstDamage bool
+}
+
+type Ghost struct {
+	Enemy     Enemy
+	IsVisible bool
+}
+
+type Ogre struct {
+	Enemy     Enemy
+	IsResting bool
+}
+
+type SnakeMage struct {
+	Enemy Enemy
+}
+
+func NewZombie(box Box) *Zombie {
+	enemy := Zombie{
+		Enemy: Enemy{
+			Character: Character{
+				Shape:    box,
+				Agility:  uint(AttributeRateLow),
+				Strength: uint(AttributeRateAverage),
+				Health:   float64(AttributeRateHigh),
+			},
+			Type:            EnemyTypeZombie,
+			HostilityRadius: HostilityRadiusAverage,
+			Direction:       DirectionStop,
+		},
+	}
+
+	return &enemy
+}
```

Comment:
Можно чутка упростить функцию:

```suggestion
func NewZombie(box Box) *Zombie {
	return &Zombie{
		Enemy: Enemy{
			Character: Character{
				Shape:    box,
				Agility:  uint(AttributeRateLow),
				Strength: uint(AttributeRateAverage),
				Health:   float64(AttributeRateHigh),
			},
			Type:            EnemyTypeZombie,
			HostilityRadius: HostilityRadiusAverage,
			Direction:       DirectionStop,
		},
	}
}
```

И аналогично для других подобных функций.